lint/flagparser/flag_parser.go

```diff
@@ -0,0 +1,84 @@
+package flagparser
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+)
+
+const EnableAll = "all"
+const DisableAll = "all"
+
+func NewFlagParser() *FlagParser {
+	fp := &FlagParser{}
+
+	flag.StringVar(&fp.Enable, "enable", EnableAll,
+		`comma-separated list of enabled checkers`)
+	flag.StringVar(&fp.ConfigFile, "config", "",
+		`name of JSON file containing checkers configurations`)
+	flag.StringVar(&fp.Disable, "disable", "",
+		`comma-separated list of disabled checkers`)
+	flag.BoolVar(&fp.WithExperimental, `withExperimental`, false,
+		`only for -enable=all, include experimental checks`)
+	flag.BoolVar(&fp.WithOpinionated, `withOpinionated`, false,
+		`only for -enable=all, include very opinionated checks`)
+	flag.IntVar(&fp.FailureExitCode, "failcode", 1,
+		`exit code to be used when lint issues are found`)
+	flag.BoolVar(&fp.CheckGenerated, "checkGenerated", false,
+		`whether to check machine-generated files`)
+	flag.BoolVar(&fp.ShorterErrLocation, "shorterErrLocation", true,
+		`whether to replace error location prefix with $GOROOT and $GOPATH`)
+
+	return fp
+}
+
+type FlagParser struct {
+	Enable             string
+	Disable            string
+	ConfigFile         string
+	WithExperimental   bool
+	WithOpinionated    bool
+	FailureExitCode    int
+	CheckGenerated     bool
+	ShorterErrLocation bool
+}
+
+func (fp *FlagParser) EnabledCheckers() []string {
+	return strings.Split(fp.Enable, ",")
+}
+
+func (fp *FlagParser) DisabledCheckers() []string {
+	return strings.Split(fp.Disable, ",")
+}
+
+func (fp *FlagParser) Parse() error {
+	flag.Parse()
+
+	if fp.Enable != EnableAll {
```

Comment:
This if can be inverted and everything can be rewritten as a simple swithch 😉

Reply:
Can you give details?

You mean something like that:
```
	if fp.Enable == EnableAll {
		return nil
	}

	switch {
	case fp.WithExperimental:
		return fmt.Errorf("-withExperimental used with -enable=%q", fp.Enable)
	case fp.WithOpinionated:
		return fmt.Errorf("-withOpinionated used with -enable=%q", fp.Enable)
	}

	return nil
```
? 

I thought about that. It give more room to made a mistake for developer who want to add more checks, he will see this switch, and probably miss that there is "return nil" before and will just expand switch with new case. Such change on my mind made code more fragile against changes.

What do you think?